### ISSUE TITLE
Wayland session prevents alacritty spawning on X11 session

### ISSUE THREAD

**Issue by L-as** [alacritty/alacritty#1499]
When logging into a second TTY and running a Wayland session from there, I can no longer run alacritty in my X11 session. It ends in a panic as such:

```
wl_drm@32: error 0: authenicate failed
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 71, kind: Other, message: "Protocol error" }', libcore/result.rs:983:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:227
             at libstd/panicking.rs:475
   3: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   4: rust_begin_unwind
             at libstd/panicking.rs:325
   5: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   6: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:26
   7: winit::window::<impl winit::WindowBuilder>::build
             at /checkout/src/libcore/result.rs:782
             at /home/las/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.15.1/src/window.rs:159
   8: glutin::GlWindow::new
             at /home/las/.cargo/registry/src/github.com-1ecc6299db9ec823/glutin-0.16.0/src/platform/linux/wayland.rs:23
             at /home/las/.cargo/registry/src/github.com-1ecc6299db9ec823/glutin-0.16.0/src/lib.rs:337
   9: alacritty::display::Display::new
             at src/window.rs:212
             at src/window.rs:234
             at /checkout/src/libcore/result.rs:691
             at src/window.rs:233
             at src/display.rs:140
  10: alacritty::main
             at src/main.rs:101
             at src/main.rs:59
  11: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  12: main
  13:   ##__libc_start_main
  14: _start
```

I have not set the `WAYLAND_DISPLAY` environment variable, only `DISPLAY` is set.

If I run 2 Wayland sessions, I can still open alacritty in both.
Running the Wayland session within the X11 session does not allow alacritty to run in the X11 session, but instead opens it in the Wayland session.

I am using commit 1adb5cb7fc05054197aa08e0d1fa957df94888ab and installed it through `cargo install --git ...`

I am unsure on how to generate a coredump.

**Comment by aereaux** [comment]
This is actually due to a "bug" in libwayland where if the WAYLAND_DISPLAY variable is not set, it tries wayland-0 anyways.  My suggestion would be either to patch libwayland to not do that, or to configure your wayland compositor to use a different wayland socket.

**Comment by chrisduerr** [comment]
Assuming that @aereaux is correct, there doesn't seem to be anything Alacritty can do about this.

I'm going to close this issue for now, if anyone has evidence that this is something Alacritty should solve, please let me know and I will re-open.

**Comment by soloturn** [comment]
@chrisduerr please reopen, this is not considered a bug in libwayland but works as designed, says at least jonas ahdal here:
https://gitlab.gnome.org/GNOME/gtk/issues/1390#note_341353

there are different variables depending on what library is used to control the behaviour, e.g. for QT and gnome / GDK:
```
QT_QPA_PLATFORM=wayland
GDK_BACKEND=x11
```

**Comment by kchibisov** [comment]
this bug should be fixed upstream since v0.4.0 IIRC
